### PR TITLE
Implement Chen+24 particle spray model

### PR DESCRIPTION
**Pull request by ybillchen**
### Describe your changes

This branch implements the new Chen+24 ([arXiv:2408.01496](https://arxiv.org/abs/2408.01496)) model for stream modeling. A [notebook](https://github.com/ybillchen/particle_spray) demonstrating how to use the model is available.

-Bill Chen

### Checklist

* [x] Did you add tests?
* [x] Did you add documentation for your changes?
* [x] Did you reference any relevant issues?
* [x] Did you add a changelog entry? (see `CHANGES.rst`)
* [x] Are the CI tests passing?
* [x] Is the milestone set?

### CODE REVIEW

**Comment by adrn** [comment]
Thanks @ybillchen -- this is looking great! I fixed some issues related to numpy 2.0, so to get the tests to pass you will need to rebase this PR. Let me know if you want help doing that!

**Comment by adrn** [comment]
I went ahead and did the rebase for you!

File: CHANGES.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ New Features
 
 - Added the Burkert potential as a built-in cpotential.
 
+- Added new particle spary method by Chen et al. (2024).
```
Comment (adrn) [inline CHANGES.rst:12]:
```suggestion
- Added new particle spray method by Chen et al. (2024).
```

**Comment by adrn** [comment]
Thanks @ybillchen!

**Comment by ybillchen** [comment]
Thank you Adrian! Super happy that this is merged to gala!

**Pull request merged**